src/meta/src/hummock/compaction/tier_compaction_picker.rs

```diff
@@ -148,6 +152,25 @@ impl TierCompactionPicker {
         true
     }
 
+    fn check_key_range_overlap(
+        &self,
+        left: &risingwave_pb::hummock::KeyRange,
+        right: &risingwave_pb::hummock::KeyRange,
+    ) -> bool {
+        // TODO: `SstableInfo` is to meet OverlapStrategy's signature. Refactor it.
```

Comment:
+1. We need to check overlap for the following cases:
1. Check whether a SST overlaps with the compacting SSTs. Skip the SST when overlap=true.
2. Check whether a SST overlaps with the selecting SSTs. Include the SST when overlap=true.

Looks like the OverlapStrategy should not be checking overlaps beween two SSTs, but checking overlaps between a SST and a compaction task (either an ongoing task or the one being generated).

Reply:
But we may check sst by hash-bucket ? So I use a strategy and  pass sstableinfo rather than keyrange